urlgrabber/grabber.py

```diff
@@ -1213,6 +1222,10 @@ def urlgrab(self, url, filename=None, opts=None, **kwargs):
         (url,parts) = opts.urlparser.parse(url, opts)
         (scheme, host, path, parm, query, frag) = parts
         opts.find_proxy(url, scheme)
+        
+        if opts.logspec is not None:
+            opts.set_loglevel(opts.logspec)
+            
```

Comment:
I would consider adding this also to the other public methods of `URLGrabber` class: `urlopen` and `urlread`.